modules/regular_service_perimeter/variables.tf

```diff
@@ -28,22 +30,24 @@ variable "perimeter_name" {
 
 variable "restricted_services" {
   description = "GCP services that are subject to the Service Perimeter restrictions. Must contain a list of services. For example, if storage.googleapis.com is specified, access to the storage buckets inside the perimeter must meet the perimeter's access restrictions."
-  type        = "list"
-  default     = []
+  type        = list(string)
+  default     = null
```

Comment:
IMHO will be better to use as default [] for list

Reply:
Thank you, Ihor!
[Resolved](https://github.com/terraform-google-modules/terraform-google-vpc-service-controls/commit/63c5a8a98dbafd8436547a6d994293cafb99c825)